platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/MerkleStateRoot.java

```diff
@@ -1042,4 +1045,24 @@ private WritablePlatformStateStore writablePlatformStateStore() {
         }
         return store;
     }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void calculateHash() {
+        throwIfMutable("Hashing should only be done on immutable states");
+        throwIfDestroyed("Hashing should not be done on destroyed states");
+        if (getHash() != null) {
+            return;
+        }
+        try {
+            MerkleCryptoFactory.getInstance().digestTreeAsync(this).get();
+        } catch (final ExecutionException e) {
+            logger.fatal(EXCEPTION.getMarker(), "Exception occurred during hashing", e);
```

Comment:
If we log a fatal exception, we should do more (e.g. throw a runtime exception, or error, or take some other action to either recover or stop the system).

Reply:
It's not that dramatic in this case, so I'm going to change `fatal` to `error`